### PR TITLE
One click to survey ui

### PR DESCRIPTION
**Pull request by mehalshah**
Put buttons on the workshop dashboard so users can go straight to viewing survey results for one of the workshops

### CODE REVIEW

**Comment by mehalshah** [comment]
image

**Comment by mehalshah** [comment]
Now there's a URL parameter that will preload a workshop

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx [hunk 1]
```diff
@@ -86,24 +87,54 @@ const SurveyResultsHeader = React.createClass({
     const courses = this.getCoursesForWorkshops();
 
     if (courses.length > 0) {
-      this.filterWorkshops(courses[0]);
+      if (this.props.preselectedWorkshopId) {
+        const preselectedWorkshop = _.find(this.props.workshops, workshop => {
+          return workshop.id === parseInt(this.props.preselectedWorkshopId);
+        });
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx:94]: nit: No need for lodash here. You can use javascript Array.find:

```ES6
const preselectedWorkshop = this.props.workshops.find(workshop =>
  workshop.id === parseInt(this.props.preselectedWorkshopId)
);
```

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx [hunk 2]
```diff
@@ -86,24 +87,54 @@ const SurveyResultsHeader = React.createClass({
...
+
+        if (preselectedWorkshop) {
+          this.filterWorkshops(preselectedWorkshop.course, preselectedWorkshop);
+        } else {
+          this.filterWorkshops(courses[0]);
+        }
+      } else {
+        this.filterWorkshops(courses[0]);
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx:100]: This is redundant and a bit tough to read. How about something like:

```ES6
let preselectedWorkshop = undefined;

if (this.props.preselectedWorkshopId) {
  preselectedWorkshop = ...
}
if (preselectedWorkshop) {
  this.filterWorkshops(preselectedWorkshop.course, preselectedWorkshop);
} else {
  this.filterWorkshops(courses[0]);
}

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx [hunk 3]
```diff
@@ -86,24 +87,54 @@ const SurveyResultsHeader = React.createClass({
...
+
+      this.setSurveyPanel(course, preselectedWorkshop.id, this.getWorkshopFriendlyName(preselectedWorkshop));
+    } else {
+      let firstWorkshopId = this.props.organizerView ? undefined : (filteredWorkshops[0] ? filteredWorkshops[0].id : undefined);
+
+      this.setState({
+        selectedCourse: course,
+        filteredWorkshops: filteredWorkshops,
+        selectedWorkshopId: firstWorkshopId,
+      });
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx]: This is also a bit hard to read. I'd prefer a single setState. It shouldn't need to know about preselectedWorkshop. That should only dictate which workshop is selected. Course and filteredWorkshops don't change, so it doesn't need to repeat.

The logic here to select the first workshop is a bit complex, but maybe something like:
```ES6
let selectedWorkshopId;
if (preselectedWorkshop) {
  selectedWorkshopId = preselectedWorkshop.id;
else {
  selectedWorkshopId = this.props.organizerView ? undefined : (filteredWorkshops[0] ? filteredWorkshops[0].id : undefined);
}

this.setState({
  selectedCourse: course,
  filteredWorkshops,
  selectedWorkshopId
});
```

File: apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx [hunk 4]
```diff
@@ -45,11 +45,21 @@ const WorkshopDashboard = (
         component={SurveyResults}
       />
       <Route
+        path="survey_results(/:workshopId)"
+        breadcrumbs="Workshops"
+        component={SurveyResults}
+      />
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx:46]: What is this new route? How is it different from the previous `survey_results` route? Are you sure you want the breadcrumbs to just display `Workshops`?

I think you want a single (combined) route here, with the optional param from line 48 and the breadcrumbs from line 44.

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx (hunk 1 above)
Comment (mehalshah) [reply apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx:94]: Done

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx (hunk 2 above)
Comment (mehalshah) [reply apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx:100]: Done

File: apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx [hunk 5]
```diff
@@ -85,18 +85,17 @@ const SurveyResultsHeader = React.createClass({
 
   componentDidMount() {
     const courses = this.getCoursesForWorkshops();
+    let preselectedWorkshop;
 
     if (courses.length > 0) {
       if (this.props.preselectedWorkshopId) {
-        const preselectedWorkshop = _.find(this.props.workshops, workshop => {
+        preselectedWorkshop = this.props.workshops.find(workshop => {
           return workshop.id === parseInt(this.props.preselectedWorkshopId);
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx]: I just noticed this: make sure you specify the radix 10 in [parseInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt):

"Always specify this parameter to eliminate reader confusion and to guarantee predictable behavior. Different implementations produce different results when a radix is not specified, usually defaulting the value to 10."

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx [hunk 6]
```diff
@@ -98,7 +98,7 @@ const WorkshopManagement = React.createClass({
         href={this.context.router.createHref(this.props.surveyUrl)}
         onClick={this.handleSurveyClick}
       >
-        Survey Results
+        View Survey Results
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx:101]: Make sure this doesn't look too cluttered with all the new text.

**Pull request merged**